src/SUFST/inc/watchdog.h

```diff
@@ -0,0 +1,30 @@
+
+#ifndef WATCHDOG_H
+#define WATCHDOG_H
+#include <stdint.h>
+#include <tx_api.h>
+
+#define WATCHDOG_TX_QUEUE_SIZE 10
```

Comment:
I'm not 100% sure if "watchdog" is the right name for this. The behaviour is really quite different from a conventional watchdog timer, which could be confusing? Last year with the VCU we had a similarly named service, and I could never decide what to call it.

Reply:
How about `exception handler` or just `fail detector`?

Reply:
Probably `error_handler` I'd say? If you're happy with that!

Reply:
CubeMX generates a function called `Error_Handler` which achieves much and such the same thing, so this could make sense, it's just a ThreadX integrated and more advanced version of it.